File: binary_search.py
# Jacobus Burger (2023)
# binary search on an array of values, O(log n) time
# see: https://en.wikipedia.org/wiki/Binary_search_algorithm
def search(array, target):
    left, right = 0, len(array) - 1
    while left <= right:
        mid = (left + right) // 2
        if target == nums[mid]:
            return mid
        elif target >= nums[mid]:
            left = mid + 1
        elif target < nums[mid]:
            right = mid - 1
    return -1

File: binary_search.c
// Jacobus Burger (2023)
// binary search on an array of values, O(log n) time
// see: https://en.wikipedia.org/wiki/Binary_search_algorithm
int
search(int target, int *array, int length)
{
        int left = 0, right = length - 1;
        while (left <= right)
        {
                int middle = (left + right) / 2;
                if (target == array[middle])
                {
                        return middle;
                }
                else if (target > array[middle])
                {
                        left = middle + 1;
                }
                else if (target < array[middle])
                {
                        right = middle - 1;
                }
        }
        return -1;
}
